packages/demo-gatsby/src/templates/blog-post.js

```diff
@@ -189,24 +189,22 @@ function BlogPostTemplate(props) {
               sticky="74px"
               name="rawMarkdownBody"
               imageProps={{
-                async upload(files) {
-                  const directory = "/static/images/"
+                // Generate the frontmatter value based on the filename
+                parse: media => `./${media.filename}`,
 
-                  let media = await cms.media.store.persist(
-                    files.map(file => ({
-                      directory,
-                      file,
-                    }))
-                  )
+                // Decide the file upload directory for the post
+                uploadDir: blogPost => {
+                  const postPathParts = blogPost.fileRelativePath.split("/")
```

Comment:
could replace this whole function w/ a call to `path.dirname` https://nodejs.org/api/path.html#path_path_dirname_path